fix: fix missing directory ctd with DynamicCubeMap Creator

langfod opened the pull request:
When a destination folder for the DynamicCubeMap Creator export does not exist then CS should not crash.

Now:
Create Folder if destination folder does not exist. Unable to create missing folder: log error
File Object with same name exists: skips

langfod committed:
DynamicCubeMap Creator crash fix

When a destination folder for the DynamicCubeMap Creator export does not exist then CS should not crash.

Now:
Create Folder if destination folder does not exist.
Unable to create missing folder: log error
File Object with same name exists: skips

langfod committed:
style: 🎨 apply clang-format changes

doodlum commented:
LGTM

alandtse commented on src/Features/DynamicCubemaps.cpp:
```diff
@@ -91,11 +91,28 @@ void DynamicCubemaps::DrawSettings()
 
-					std::wstring wfilename = std::wstring(filename.begin(), filename.end());
-					DX::ThrowIfFailed(SaveToDDSFile(image.GetImages(), image.GetImageCount(), image.GetMetadata(), DirectX::DDS_FLAGS::DDS_FLAGS_NONE, wfilename.c_str()));
+					if (defaultDynamicCubeMapSavePathSafe) {
```
Shouldn't it be just a check if it exists?

langfod replied:
my thought was to not not attempt saving the file if there was a failure in creating a new directory, but allow the rest of the method to continue (.Release etc...)

Was not sure how to use Imgui to inform user, which means silently fails in the UI.

or would it be preferable to just wrap the entire thing in a directory check and do nothing but log if the directory does not exist?

alandtse replied:
I'm thinking the simplest way is to look at the use case. A user is trying to just save a cubemap. So we first solve the CTD, like you mentioned, and then if they try to save and it fails, just some sort of messaging. See the message popup we do for failed shaders.

alandtse replied:
See https://github.com/doodlum/skyrim-community-shaders/blob/dev/src/Menu.cpp#L195-L208

langfod replied:
kept the directory creation for now. 
perhaps will makes this configurable in a later change.

saves will not be attempted if a file object with the same name exists. 
generic message popup  added as Issue #519 

currently message error message display is tightly coupled with ShaderCompiler

alandtse commented on src/Features/DynamicCubemaps.cpp:
```diff
@@ -91,11 +91,28 @@ void DynamicCubemaps::DrawSettings()
+							logger::info("DynamicCubeMap Creator file for {} already exists, skipping.", filename.string());
+						} else {
+							DX::ThrowIfFailed(SaveToDDSFile(image.GetImages(), image.GetImageCount(), image.GetMetadata(), DirectX::DDS_FLAGS::DDS_FLAGS_NONE, DynamicCubeMapSavePath.c_str()));
+							logger::info("DynamicCubeMap Creator file for {} written", filename.string());
```
Do we need a try for saving errors?

langfod replied:
will try to force an error on that to see. 

I did not note DX::ThrowIfFailed in try blocks which I did a very quick look.

alandtse replied:
I believe it is catchable, I'd try first with a filename that isn't writeable.

langfod replied:
related DX:ThrowIfFailed moved inside try/catch block

alandtse commented:
@langfod Please also approve #513.

langfod committed:
Merge branch 'doodlum:dev' into dyncubemapsavefix

langfod committed:
Update DynamicCubemaps.cpp

Moved Texture creation and SaveToDDSFile inside try/catch block.

This removed one boolean variable that had been used .
No attempt is made to save a file if that filename exists (even an incorrect file object).

As there are multiple types of exceptions including OS filesystem errors the catch was left as std::exception instead of multiple catch types and specific log messages.

alandtse commented:
Also once you've addressed comments, close them out and let us know this is ready to go.

langfod commented:
Logs actions and errors.
Should not cause CTD's

Future: Add UI level notification
